### PR TITLE
libMesh submodule update

### PR DESCRIPTION
**Pull request by roystgnr**
- Support for reading (the surface triangulations defined by) STL files.  This combined with our recent tetrahedralization support also allows for creating volume meshes from STL files; however, creating high-quality volume meshes from arbitrary STL files is still a work in progress.
- Support for higher-order (up to quintic) vector-valued FE bases in the Nédélec (first kind) and Ravier-Thomas FE families.
- Better handling for odd IsoGeometric Analysis extraction operators, and better error handling for broken IGA operators.
- Better unit test coverage for NodeElem; fixes to NodeElem methods which should be no-op but instead threw errors.
- Unit test coverage for fifth-order `MONOMIAL` elements.  So far new tests have not uncovered any problems, but based on user reports of convergence issues we recommend limiting `MONOMIAL` variables to fourth-order.
- Fixes for condensed eigensolve support after mesh reinit
- Fixes for caching of FE data on sides of meshes with non-uniform p-refinement levels
- Minor fixes for compiler complaints from `nvc++` and `clang++ -fsanitize`
- Properly register `--keep-cout` and `--drop-cerr` warnings with PETSc to avoid "options you set that were not used" false positives.
- Better error messages when tetrahedralizing a non-watertight input boundary mesh.
- Refactoring in AbaqusIO, to enable subclasses in MOOSE code with experimental support for more advanced file features.
- Improvements and fixes in Reduced Basis code, particularly the Empirical Interpolation Method classes.

### CODE REVIEW

**Comment by loganharbour** [comment]
Roy adds-all-the-tests Stogner. Look at this guy

**Comment by roystgnr** [comment]
Yup, worked fine after the server did.

File: conda/libmesh/meta.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # As well as any directions pertaining to modifying those files.
 # ALSO: Follow the directions in scripts/tests/versioner_hashes.yaml
 {% set build = 1 %}
```
Comment (milljm) [inline conda/libmesh/meta.yaml:7]:
```suggestion
# ALSO: Follow the directions in scripts/tests/versioner_hashes.yaml
{% set build = 0 %}
```

File: conda/moose-dev/conda_build_config.yaml [hunk 2]
```diff
@@ -3,8 +3,8 @@ mpi:
   - openmpi
 
 moose_libmesh:
-  - moose-libmesh 2024.08.30 mpich_1
-  - moose-libmesh 2024.08.30 openmpi_1
+  - moose-libmesh 2024.10.17 mpich_1
+  - moose-libmesh 2024.10.17 openmpi_1
```
Comment (milljm) [inline conda/moose-dev/conda_build_config.yaml:7]:
```suggestion
  - moose-libmesh 2024.10.17 mpich_0
  - moose-libmesh 2024.10.17 openmpi_0
```

**Review by milljm — CHANGES_REQUESTED** [review]
Unfortunately we forgot to reset the build number... Sorry to be the bearer of bad news!

**Review by milljm — APPROVED** [review]
For my part, Conda changes look good.

**Comment by lindsayad** [comment]
I would like to get the static condensation stuff into MOOSE fairly soon. I wouldn't delay this PR if we're fine doing another update soon

**Comment by roystgnr** [comment]
I'm good with another update in a couple weeks.

**Comment by loganharbour** [comment]
Please check if next is broken or not before merging environment changes

**Pull request merged**